### PR TITLE
Don't use for..in for array iteration

### PR DESCRIPTION
**Pull request by bobbrow**
This fixes an issue while trying to set a build preset.  for..in should not be used when iterating through an array.  It will return back other properties (e.g. attached functions) that you don't intend to index out of the array object.

### CODE REVIEW

File: src/taskprovider.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 import * as vscode from 'vscode';
-import { isString } from 'util';
+import {isString} from './util';
```
Comment (bobbrow) [inline src/taskprovider.ts:6]: Saw this while searching for `for..in` usages in the codebase so I just bundled it in.  We were using the wrong `isString`.

**Review by andreeis — APPROVED** [review]

File: src/preset.ts [hunk 2]
```diff
@@ -878,13 +878,13 @@ async function expandBuildPresetHelper(preset: BuildPreset,
     if (util.isString(preset.targets)) {
       preset.targets = await expandString(preset.targets, expansionOpts);
     } else {
-      for (const index in preset.targets) {
+      for (let index = 0; index < preset.targets?.length; index++) {
```
Comment (xisui-MSFT) [inline src/preset.ts:881]: else if (preset.targets), and remove all the `?`s in this change.

**Review by xisui-MSFT — CHANGES_REQUESTED** [review]

File: src/preset.ts (hunk 2 above)
Comment (xisui-MSFT) [reply src/preset.ts:881]: actually it's already checked (hidden in compare view...). Can you remove all the '?'s in your changes?

Comment (bobbrow) [reply src/preset.ts:881]: Yes. I noticed the same. We've already validated the types of the presets file at this point, right?  I don't need to make sure it's an array or anything do I?

Comment (xisui-MSFT) [reply src/preset.ts]: That's correct.

**Review by xisui-MSFT — APPROVED** [review]

**Pull request merged**